### ISSUE TITLE
AsyncMapResult not iterable when not ready

### ISSUE THREAD

**Issue by IAmLFish** [ipython/ipyparallel#274]
I am working with:
* Windows 10
* Python 3.5.3
* Jupyter Notebook 4.4.0
* ipyparallel 6.0.2

and I refer to the following docs:
http://ipyparallel.readthedocs.io/en/latest/asyncresult.html#map-results-are-iterable

I am kinda confused and desperate how one should work with AsyncMapResult.
Here is my example I want to have as a MWE (slightly changed from the docs)
```python
import ipyparallel as ipp
rc = ipp.Client()
lview = rc.load_balanced_view()

def sleep_here(t):
    import time
    time.sleep(t)
    return t

# one call per task
print("running with one call per task")
amr = lview.map(sleep_here, [.01*t for t in range(30)])
for i,r in enumerate(amr):
    print("task %i: %.3f" % (i, r))
```
but whenever I execute it I get two exceptions `TimeoutError` and `OverflowError`:
```
TimeoutError: Result not ready.

During handling of the above exception, another exception occurred:

OverflowError                             Traceback (most recent call last)
<ipython-input-7-2f80cb2e89c6> in <module>()
     11 print("running with one call per task")
     12 amr = lview.map(sleep_here, [.01*t for t in range(30)])
---> 13 for i,r in enumerate(amr):
     14     print("task %i: %.3f" % (i, r))

C:\Users\IAmLFish\Anaconda3\lib\site-packages\ipyparallel\client\asyncresult.py in __iter__(self)
    671     def __iter__(self):
    672         it = self._ordered_iter if self.ordered else self._unordered_iter
--> 673         for r in it():
    674             yield r
    675 

C:\Users\IAmLFish\Anaconda3\lib\site-packages\ipyparallel\client\asyncresult.py in _ordered_iter(self)
    695             evt = Event()
    696             for child in self._children:
--> 697                 self._wait_for_child(child, evt=evt)
    698                 for r in self._yield_child_results(child):
    699                     yield r

C:\Users\IAmLFish\Anaconda3\lib\site-packages\ipyparallel\client\asyncresult.py in _wait_for_child(child, evt, timeout)
    375         evt.clear()
    376         child.add_done_callback(lambda f: evt.set())
--> 377         evt.wait(timeout)
    378 
    379     # asynchronous iterator:

C:\Users\IAmLFish\Anaconda3\lib\threading.py in wait(self, timeout)
    547             signaled = self._flag
    548             if not signaled:
--> 549                 signaled = self._cond.wait(timeout)
    550             return signaled
    551 

C:\Users\IAmLFish\Anaconda3\lib\threading.py in wait(self, timeout)
    295             else:
    296                 if timeout > 0:
--> 297                     gotit = waiter.acquire(True, timeout)
    298                 else:
    299                     gotit = waiter.acquire(False)

OverflowError: timeout value is too large
```
Furthermore, the docs state

> That is to say, if you treat an AsyncMapResult as if it were a list of your actual results, it should behave as you would expect, with the only difference being that you can start iterating through the results before they have even been computed.

Unfortunately, I cannot confirm this.
Therefore, why is this such a big issue? Do I have to wrap the `amr` or how can I iterate over the finished jobs while other jobs are still computing?

I do not want to use `lview.map_sync` for my case and definitely want to work with "AsyncMapResult" but I can't figure out if I do something wrong or if there is/are some bug(s) involved.

**Comment by michaelmcroberts** [comment]
I believe this is caused by client/asyncresult.py using a value of 1e8 for the wait timeout. According to [Python 3 Threading](https://docs.python.org/3/library/threading.html) the max value for a timeout should not exceed threading.TIMEOUT_MAX or an OverflowError will be thrown. On my configuration this value was 4294967. I believe this would be resolved by Issue #269

**Comment by minrk** [comment]
Fixed by #269

**Comment by IAmLFish** [comment]
@minrk Sorry, I cannot confirm this yet, I get the same error. Did you run my MWE above on a jupyter notebook? 
(Windows 10, Python 3.5.3 and recent version of IPython, Jupyter, ipyparallel)

**Comment by nlooije** [comment]
@minrk @IAmLFish I also cannot confirm this fixed. Running W10, Python 3.6.2 (from Anaconda 3 distribution)

**Comment by minrk** [comment]
This hasn't been released yet. It will be fixed in 6.1
